src/release.js

```diff
@@ -66,22 +67,31 @@ module.exports = async function ({ github, context, inputs }) {
 
   const opticToken = inputs['optic-token']
 
-  if (inputs['npm-token']) {
-    await run('npm', [
-      'config',
-      'set',
-      `//registry.npmjs.org/:_authToken=${inputs['npm-token']}`,
-    ])
-
-    await run('npm', ['pack', '--dry-run'])
-    if (opticToken) {
-      const otp = await run('curl', ['-s', `${opticUrl}${opticToken}`])
-      await run('npm', ['publish', '--otp', otp, '--tag', npmTag])
+  try {
+    if (inputs['npm-token']) {
+      await run('npm', [
+        'config',
+        'set',
+        `//registry.npmjs.org/:_authToken=${inputs['npm-token']}`,
+      ])
+
+      await run('npm', ['pack', '--dry-run'])
+      if (opticToken) {
+        const otp = await run('curl', ['-s', `${opticUrl}${opticToken}`])
+        await run('npm', ['publish', '--otp', otp, '--tag', npmTag])
+      } else {
+        await run('npm', ['publish', '--tag', npmTag])
+      }
```

Comment:
since gh is stuck and it will require another commit to trigger CI, why don't we take the chance to extract this inline logic in functions? it will make it easier to make sense of the logical flow of the code